lib/context_adapter_communication.js

```diff
@@ -32,59 +32,79 @@ var caLogger = require('logops'),
 
 /**
  * Sends a queryContext request to the configured Context Broker
- * @param {string} service Service header to be used in the request
- * @param {string} servicePath Service Path header to be used in the request
+ * @param {object} request The original HTTP request
  * @param {object} payload Payload to be included in the request
  * @param {function} callback Callback to be called informing of any errors or returning
  *  the obtained response
  */
-function sendQueryContext(service, servicePath, payload, callback) {
+function sendQueryContext(request, payload, callback) {
+  var headers = {
+    'Accept': 'application/json',
+    'Content-Type': 'application/json',
+    'Fiware-Service': request.headers[caConfig.HEADER.FIWARE_SERVICE],
+    'Fiware-ServicePath': request.headers[caConfig.HEADER.FIWARE_SERVICE_PATH]
+  };
+  if (caHelper.getCorrelator(request)) {
+    headers['Fiware-Correlator'] = caHelper.getCorrelator(request);
```

Comment:
Yeah, I just wanted to include the header it was passed in the notification request or include a new one if it was not included.